dask/array/blockwise.py

```diff
@@ -161,29 +161,28 @@ def blockwise(
             v = (v,)
         chunkss[k] = v
     arginds = list(zip(arrays, args[1::2]))
-
-    for arg, ind in arginds:
-        if hasattr(arg, "ndim") and hasattr(ind, "__len__") and arg.ndim != len(ind):
-            raise ValueError(
-                "Index string %s does not match array dimension %d" % (ind, arg.ndim)
-            )
-
-    numblocks = {a.name: a.numblocks for a, ind in arginds if ind is not None}
+    numblocks = {}
 
     dependencies = []
     arrays = []
 
     # Normalize arguments
     argindsstr = []
-    for a, ind in arginds:
+
+    for arg, ind in arginds:
+        if hasattr(arg, "ndim") and hasattr(ind, "__len__") and arg.ndim != len(ind):
+            raise ValueError(
+                "Index string %s does not match array dimension %d" % (ind, arg.ndim)
+            )
```

Comment:
It would be good to try to refactor this to avoid these `hasattr` calls. They tend to be pretty expensive when used in big loops.

Reply:
@kkraus14 - Agreed, but I wouldn't expect these loops to get "huge".  So, unless you or @galipremsagar have a simple workaround, I suggest that we not worry too much about this.

Reply:
Had a discussion offline @kkraus14 and upon Keith's idea moved the `hasattr` check to else block when `ind` is None. Thanks for the catch @kkraus14 🙏 

Would you mind having a review at the change just made @rjzamora ?